Dashboard restructed, more widgets added and code refactored

avinash403 opened the pull request:
resolves #1006

ashu555 commented on resources/views/themes/default1/order/index.blade.php:
```diff
@@ -220,8 +220,10 @@
         $('#order-table').DataTable({
```
Reset button should reset all the values to before

avinash403 replied:
didn't get you. Please elaborte

ashu555 commented on resources/views/themes/default1/order/index.blade.php:
```diff
@@ -220,8 +220,10 @@
-            order: [[ 7, "desc" ]],
+            stateSave: false,
+            // if in request sort field is present, it will take that else default order
+            // need to stringify the sort_order, else it will be considered as a javascript variable
```
What if we select version less than equal to and version greater than equal to at once. Should it be that if we select either of the one the other gets disabled or some validation of that kind?

ashu555 replied:
Similarly if we select paid products and version less than equal to (lets say 2.0.1), it should show the only the products that are paid and is less that v2.0.1. Just mentioning, don't know if it is tested or not.

ashu555 replied:
Do we need to remove the option for searching the exact version?

avinash403 replied:
> What if we select version less than equal to and version greater than equal to at once. Should it be that if we select either of the one the other gets disabled or some validation of that kind?

it is a range. It will work even if you select both

avinash403 replied:
> Similarly if we select paid products and version less than equal to (lets say 2.0.1), it should show the only the products that are paid and is less that v2.0.1. Just mentioning, don't know if it is tested or not.

yes, it will work that way

avinash403 replied:
> Do we need to remove the option for searching the exact version?

it can be acheived if you select both options as same. Though it has less use-cases, its not a good idea to give a seperate option for that

ashu555 replied:
> > Similarly if we select paid products and version less than equal to (lets say 2.0.1), it should show the only the products that are paid and is less that v2.0.1. Just mentioning, don't know if it is tested or not.
> 
> yes, it will work that way

It is showing unpaid products also even if we select paid products along with version less than dropdown

avinash403 commented:
@ashu555 mentioned changes are pushed

ashu555 commented on resources/views/themes/default1/order/index.blade.php:
```diff
@@ -99,53 +99,28 @@
-                     <button name="Reset" type="submit" id="reset" class="btn btn-danger" data-loading-text="<i class='fa fa-refresh fa-spin fa-1x fa-fw'>&nbsp;</i> updating..."><i class="fa fa-refresh">&nbsp;&nbsp;</i>{!!Lang::get('Reset')!!}</button>
-
-
+                    <a class="btn btn-danger" href="{!! url('/orders') !!}"><i class="fa fa-refresh">&nbsp;{!!Lang::get('Reset')!!}</i></a>
```
Reset Font looks the same to you?

avinash403 replied:
Thanks for pointing. Fixed

ashu555 commented on resources/views/themes/default1/order/index.blade.php:
```diff
@@ -99,53 +99,28 @@
-                {!! Form::select('version_greater_than_equal',[null => 'Select']+ array_combine($allVersions, $allVersions), $request->version_greater_than_equal,
-                ['class' => 'form-control','id'=>'version_greater_than_equal']) !!}
+                {!! Form::label('version_from','Version From') !!}
+                {!! Form::select('version_from',[null => 'Select']+ array_combine($allVersions, $allVersions), $request->version_from,
```
We are using 'Choose' instead of 'Select' in the system.

ashu555 replied:
There was an issue if you remember to disable values coming from cache in THE date range fields. If we have selected dates before for searching, it shows recommedations to select those dates. This sometimes covers the date picker. Can this be also fixed in this PR?

avinash403 replied:
> We are using 'Choose' instead of 'Select' in the system.

fixed

avinash403 replied:
> There was an issue if you remember to disable values coming from cache in THE date range fields. If we have selected dates before for searching, it shows recommedations to select those dates. This sometimes overlaps with the date picker. Can this be also fixed in this PR?

sure. fixed

SarfrazS commented:
The development branch also not working.

avinash403 commented:
> first, I checked with fresh DB after that I checked with exiting billing DB.
> In this PR i checked 2 widgets Paid Orders Expired in the Last 30 days and Clients With outdated Product Version (Last 30). I changed the expiry date manually and version and then tested. working but some link not working please check and fix it. If I did some mistakes, please let me know. below doc test case are there.
> https://docs.google.com/document/d/15qjlbJ1wD3x0eSrCtmVE0c0phIZlnofFmlRLu_h4LPs/edit

fixed asked issue
